src/azul/service/manifest_service.py

```diff
@@ -933,7 +934,11 @@ def file_name(self, object_key, base_name: Optional[str] = None) -> str:
         return file_name
 
     def tagging(self, file_name: Optional[str]) -> Optional[Mapping[str, str]]:
-        return None if file_name is None else {self.service.file_name_tag: file_name}
+        if file_name is None:
+            return None
+        else:
+            encoded_file_name = base64.urlsafe_b64encode(file_name.encode())
+            return {self.service.file_name_tag: encoded_file_name.decode('ascii')}
```

Comment:
There should be total anti-symmetry. Decoding should be the exact opposite of encoding. The fact that `'asci'` only occurs once is a red flag.